apisix/core/request.lua

```diff
@@ -68,12 +68,25 @@ function _M.header(ctx, name)
 end
 
 
-function _M.set_header(header_name, header_value)
+function _M.set_header(ctx, header_name, header_value)
+    if type(ctx) == "string" then
+        -- It would be simpler to keep compatibility if we put 'ctx'
+        -- after 'header_value', but the style is too ugly!
```

Comment:
how about print a warning message to log?

Reply:
@membphis 
Added.